src/socket.rs

```diff
@@ -368,6 +368,22 @@ impl Socket {
         self.inner.set_ttl(ttl)
     }
 
+    /// Set the value of IP_BOUND_IF option on this socket.
+    ///
+    /// See libc::if_nametoindex on mapping interface name to interface index.
+    #[cfg(all(feature = "all", not(target_os = "redox")))]
+    pub fn set_bound_interface(&self, if_index: sys::c_uint) -> io::Result<()> {
```

Comment:
We normally use `i32` instead of `sys::c_int`, so I think `u32` should be used here.